src/librustdoc/core.rs

```diff
@@ -415,6 +419,23 @@ pub fn run_core(options: RustdocOptions) -> (clean::Crate, RenderInfo, RenderOpt
 
             let mut krate = clean::krate(&mut ctxt);
 
+            if let Some(ref m) = krate.module {
+                match m.doc_value() {
+                    None | Some("") => {
+                        let mut diag = tcx.struct_lint_node(
+                            rustc_lint::builtin::NO_CRATE_LEVEL_DOC,
+                            ctxt.as_local_hir_id(m.def_id).unwrap(),
+                            "No documentation found on this crate top module.\n\n\
```

Comment:
Not sure how we really talk about the top level of a crate, is it a module per-se?

If so, perhaps:

```suggestion
                            "no documentation found for this crate's top-level module.\n\n\
```